AutoT4/ProjectItemSettings.cs

```diff
@@ -67,9 +67,24 @@ protected T Get<T>(T defaultValue = default(T), [CallerMemberName] string name =
             if (0 != _storage.GetItemAttribute(_itemId, Scope(name), out serializedValue))
                 return defaultValue;
 
-            return !String.IsNullOrWhiteSpace(serializedValue)
-                       ? (T)Convert.ChangeType(serializedValue, typeof(T))
-                       : defaultValue;
+            if (string.IsNullOrWhiteSpace(serializedValue))
+                return defaultValue;
+
+            try
+            {
+                if (typeof(T).IsEnum)
+                {
+                    return (T)Enum.Parse(typeof(T), serializedValue);
+                }
+                return (T)Convert.ChangeType(serializedValue, typeof(T));
+            }
+            catch
+            {
+                if (coercionFunc != null)
```

Comment:
I was thinking about the coercion function as the last resort, that will only get executed if `Enum.Parse` or `Convert.ChangeType` throw. When it comes to the coercion function throwing exceptions my thoughts are that who supplies it should have `try/catch` block in it (if needed), as it makes for a more predictable behavior.

Anyways, should i place it after the `if` on line 75?